### ISSUE TITLE
[bug] sync_issues tool can not handle the error caused by the transferred or deleted issue

### ISSUE THREAD

**Issue by Mini256** [cncf/devstatscode#9]
When I use [sync_issues](https://github.com/cncf/devstatscode/blob/master/cmd/sync_issues/sync_issues.go) to patch the missing issue status, I found that when it encounters an issue that has been deleted and transferred, it will cause an API error and terminate the whole program.

Maybe we should handle these error status codes:

https://docs.github.com/en/rest/reference/issues#:~:text=with%20GitHub%20Apps-,Get%20an%20issue,-The%20API%20returns

**Comment by lukaszgryglicki** [comment]
hi, this tool is not a part of the standard DevStats workflow, can you at least provide an exact error stack trace or steps to replicate this (command line, repo, etc.)?
PRs of course welcome.

**Comment by Mini256** [comment]
OK, I will submit the error stack in my spare time and submit PR to fix the problem.

**Comment by lukaszgryglicki** [comment]
Great thanks, you can just put the stack and/or example of how to replicate. PR is welcomed, but if you don't have a time to fix this - I can try to do that too, just let me know.

**Comment by Mini256** [comment]
This is the case of a deleted issue：[pingcap/tiup#455 ](https://github.com/pingcap/tiup/issues/455)

```
Compiled None, commit: None on None using None
2021-08-28 23:37:05 /___4sync_issues: sync_issues.go: Processing 1 issues - GHAPI part
2021-08-28 23:37:44 /___4sync_issues: /private/var/folders/0g/2fs5nb9j7z5bl5p2r_kjckg00000gn/T/GoLand/___4sync_issues error: *github.ErrorResponse:GET https://api.github.com/repos/pingcap/tiup/issues/455: 410 This issue was deleted [], non fatal, exiting 0 status
Exiting.
```

This is the case of a transferred issue：[pingcap/tidb#13706](https://github.com/pingcap/tidb/issues/13706)

```
/private/var/folders/0g/2fs5nb9j7z5bl5p2r_kjckg00000gn/T/GoLand/___8sync_issues
Compiled None, commit: None on None using None
2021-08-29 20:17:19 /___8sync_issues: sync_issues.go: Processing 1 issues - GHAPI part
2021-08-29 20:17:20 /___8sync_issues: /private/var/folders/0g/2fs5nb9j7z5bl5p2r_kjckg00000gn/T/GoLand/___8sync_issues error: *url.Error:Get "https://api.github.com/repos/pingcap/tidb/issues/13707": 301 response missing Location header, non fatal, exiting 0 status
```

The problem of deleted issues is easier to solve. We can skip the deleted issues directly.，fix on #10

The problem of the transferred issue seems to be more difficult to solve, because it may be caused by the rename of the repository, or it may be due to the issue being migrated to another repository.

Especially for the use case of `pingcap/tidb#13706`, it was transferred to another private repository (pingcap/example). 

- If the token has permission to access, the github client will return code `200` and a new issue number (pingcap/example#220).
- If the token does not have access rights, it will return a 301 status code.
